### PR TITLE
xtest: optionally skip some slow tests

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Moves some slow tests to level 1 instead of level 0 to speed up the
default tests.

Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

### CODE REVIEW

File: host/xtest/xtest_4000.c [hunk 1]
```diff
@@ -2964,195 +2964,231 @@ struct xtest_ac_case {
...
 			  ac_rsassa_vect10, ARRAY, WITHOUT_SALT),
-	XTEST_AC_RSA_CASE(TEE_ALG_RSASSA_PKCS1_V1_5_SHA512, TEE_MODE_SIGN,
+	XTEST_AC_RSA_CASE(1, TEE_ALG_RSASSA_PKCS1_V1_5_SHA512, TEE_MODE_SIGN,
 			  ac_rsassa_vect11, ARRAY, WITHOUT_SALT),
-	XTEST_AC_RSA_CASE(TEE_ALG_RSASSA_PKCS1_V1_5_SHA512, TEE_MODE_VERIFY,
+	XTEST_AC_RSA_CASE(1, TEE_ALG_RSASSA_PKCS1_V1_5_SHA512, TEE_MODE_VERIFY,
 			  ac_rsassa_vect11, ARRAY, WITHOUT_SALT),
 
-	XTEST_AC_RSA_CASE(TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA1, TEE_MODE_VERIFY,
+	XTEST_AC_RSA_CASE(0, TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA1,
```
Comment (jbech-linaro) [inline host/xtest/xtest_4000.c]: Better to make SHA1 level 1 here  (and on line 3135) ...

File: host/xtest/xtest_4000.c [hunk 2]
```diff
@@ -2964,195 +2964,231 @@ struct xtest_ac_case {
...
-	XTEST_AC_RSA_CASE(TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA224, TEE_MODE_VERIFY,
+	XTEST_AC_RSA_CASE(1, TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA224,
+			  TEE_MODE_VERIFY,
 			  ac_rsassa_vect17, ARRAY, WITH_SALT),
-	XTEST_AC_RSA_CASE(TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA224, TEE_MODE_SIGN,
+	XTEST_AC_RSA_CASE(1, TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA224,
+			  TEE_MODE_SIGN,
 			  ac_rsassa_vect17, ARRAY, WITH_SALT),
-	XTEST_AC_RSA_CASE(TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA256, TEE_MODE_VERIFY,
+	XTEST_AC_RSA_CASE(1, TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA256,
```
Comment (jbech-linaro) [inline host/xtest/xtest_4000.c]: ... and make the SHA-256 here level 0 instead (and on line 3146). Rationale, SHA-256 have replace SHA-1 as the default hash nowadays.

File: host/xtest/xtest_4000.c [hunk 3]
```diff
@@ -3289,8 +3325,8 @@ static const struct xtest_ac_case xtest_ac_cases[] = {
 	XTEST_AC_DSA_CASE(15, TEE_ALG_DSA_SHA224, TEE_MODE_VERIFY, ac_dsa_vect180),
 	XTEST_AC_DSA_CASE(15, TEE_ALG_DSA_SHA224, TEE_MODE_SIGN, ac_dsa_vect180),
 	/* [mod = L=2048, N=256, SHA-256] */
-	XTEST_AC_DSA_CASE(0, TEE_ALG_DSA_SHA256, TEE_MODE_VERIFY, ac_dsa_vect181),
-	XTEST_AC_DSA_CASE(0, TEE_ALG_DSA_SHA256, TEE_MODE_SIGN, ac_dsa_vect181),
+	XTEST_AC_DSA_CASE(1, TEE_ALG_DSA_SHA256, TEE_MODE_VERIFY, ac_dsa_vect181),
```
Comment (jbech-linaro) [inline host/xtest/xtest_4000.c]: Would probably be good to have this enable to at least have one DSA case verified by default.

**Comment by jbech-linaro** [comment]
Just a few recommendations, other than that:
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: host/xtest/xtest_4000.c (hunk 1 above)
Comment (jenswi-linaro) [inline host/xtest/xtest_4000.c]: OK, will fix.

File: host/xtest/xtest_4000.c (hunk 2 above)
Comment (jenswi-linaro) [inline host/xtest/xtest_4000.c]: OK, will fix.

File: host/xtest/xtest_4000.c (hunk 3 above)
Comment (jenswi-linaro) [inline host/xtest/xtest_4000.c]: DSA-SHA1 is enabled, but I'll disable that and enable this instead.

**Comment by jenswi-linaro** [comment]
Updated and tag applied.

**Pull request merged**